### ISSUE TITLE
temporary egg files are created as read only and can't be deleted with shutil on Windows

### ISSUE THREAD

**Issue by grant-humphries** [buildout/buildout#295]
I'm fetching some eggs from github repos for a project and what appears to be happening is that the python files are being downloaded to a temporary directory, zipped up, moved into the develop-eggs directory and then deleted.  However some of these temporary files are created as read only and thus `shutil` is throwing an error when it tries to delete them.  I'm using Windows 7.  [This post](http://stackoverflow.com/questions/2656322/python-shutil-rmtree-fails-on-windows-with-access-is-denied) seems to have a solution, but I wasn't sure if this could be included in the code that will run on other operating systems as some of the permission settings look Windows specific.  The full error message that I'm getting is below:

```
An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "c:\miniconda2\lib\site-packages\zc\buildout\buildout.py", line 1992, in main
    getattr(buildout, command)(args)
  File "c:\miniconda2\lib\site-packages\zc\buildout\buildout.py", line 666, in install
    installed_files = self[part]._call(recipe.install)
  File "c:\miniconda2\lib\site-packages\zc\buildout\buildout.py", line 1407, in _call
    return f()
  File "c:\dev_projects\ott\otp_client_py\eggs\zc.recipe.egg-2.0.3-py2.7.egg\zc\recipe\egg\egg.py", line 126, in install
    reqs, ws = self.working_set()
  File "c:\dev_projects\ott\otp_client_py\eggs\zc.recipe.egg-2.0.3-py2.7.egg\zc\recipe\egg\egg.py", line 84, in working_set
    allow_hosts=self.allow_hosts)
  File "c:\miniconda2\lib\site-packages\zc\buildout\easy_install.py", line 883, in install
    return installer.install(specs, working_set)
  File "c:\miniconda2\lib\site-packages\zc\buildout\easy_install.py", line 714, in install
    for_buildout_run=for_buildout_run):
  File "c:\miniconda2\lib\site-packages\zc\buildout\easy_install.py", line 565, in _get_dist
    shutil.rmtree(tmp)
  File "c:\miniconda2\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\miniconda2\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\miniconda2\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\miniconda2\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "c:\miniconda2\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "c:\miniconda2\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'c:\\users\\humphrig\\appdata\\local\\temp\\tmps3wod2get_dist\\geocoder.git\\.git\\objects\\pack\\pack-08fbd107aeb576051851637552b2200eece70c83.idx'
```

**Comment by grant-humphries** [comment]
am using buildout 2.5.1 btw

**Comment by grant-humphries** [comment]
and python 2.7.11

**Comment by k-bu** [comment]
We're having exactly the same issue happening regularly, which is kinda frustrating. While zc.buildout comes with its own rmtree function (see zc.buildout.rmtree.rmtree()), why doesn't it consequently use it, especially in the easy_install module?

**Comment by reinout** [comment]
I've released the fix by @k-bu as 2.11.5, so hopefully it is fixed now!

**Comment by k-bu** [comment]
Thx a lot!
